polyply/src/generate_templates.py

```diff
@@ -325,10 +325,8 @@ def _gen_templates(self, meta_molecule):
         """
         resnames = set(nx.get_node_attributes(meta_molecule.molecule,
                                               "resname").values())
-        templates = {}
-        volumes = {}
         for resname in resnames:
-            if not resname in self.resnames:
+            if not resname in self.resnames and resname not in meta_molecule.templates:
```

Comment:
I would suggest changing the order of the if/elif/else around a bit. First the simple cases, then the complex ones.